add shell check to pre-commit

auto shell check any shell script created (I think there is only one at this point)

Added the shellcheck-py hook and ran it against .devcontainer/setup.sh (only shell script in here) — it fails right away though. Missing shebang (SC2148) and every single line has CRLF endings (SC1017), guessing it got committed from Windows without line-ending handling.

Two options here that I'm thinking :

Just land the hook config as-is, let setup.sh fail until someone patches it later
Fix setup.sh in this same PR (shebang + strip the CRLF) so it's clean from the start

I'd lean toward option 2 so nobody's greeted with a failing hook on day one, but happy to keep this PR just the config if you'd rather split it up. Lmk!

--- a/.devcontainer/setup.sh
+++ b/.devcontainer/setup.sh
@@ -1,3 +1,4 @@
+#!/bin/sh
 python -m pip install --user --upgrade pip
 python -m pip install --user pip-tools
 python -m piptools compile --upgrade -o requirements.txt pyproject.toml

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,0 +1,1 @@
+.devcontainer/setup.sh text eol=lf

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,1 +1,1 @@
-.devcontainer/setup.sh text eol=lf
+**/*.sh text eol=lf

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -45,3 +45,7 @@ repos:
         language: system
         always_run: true
         pass_filenames: false
+  - repo: https://github.com/shellcheck-py/shellcheck-py
+    rev: v0.11.0.1
+    hooks:
+      - id: shellcheck

